### PR TITLE
Added various features related to autonomous.

### PR DESCRIPTION
**Pull request by 10tothe6**
- made it so the swerve motors go into brake mode when the bot gets disabled
- added a function that allows you to set the idle states of all swerve motors
- added 3 paths and 1 auto using FRC Pathplanner
- some documentation changes/additions

### CODE REVIEW

**Review by Timatoot — APPROVED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -94,6 +94,7 @@ public Command getAutonomousCommand() {
         return autoChooser.getSelected();
     }
 
+    //this function is called when the robot enters disabled mode, it sets the motors to brake mode
```
Comment (noah4546) [inline src/main/java/frc/robot/RobotContainer.java]: Put a space before 'this' and the '//' as well as capitalize the 'T'

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -243,6 +243,11 @@ public void resetEncoders() {
     m_rearRight.resetEncoders();
   }
 
+  /**
+   * Sets the idle states of the sparkMAX motors.
```
Comment (noah4546) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: Capitalize the 'S' in 'SparkMAX'

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -243,6 +243,11 @@ public void resetEncoders() {
...
+   *
+   * @param mode the mode to set the states to (0 is coast, 1 is brake)
+   */
   public void setIdleStates(int mode) {
     m_frontLeft.setIdle(mode);
```
Comment (noah4546) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:252]: Variables: No under_scores (except constants).

This whole class uses underscores for variable names, and should be fixed.
'm_frontLeft' should be 'mFrontLeft', 'frontLeftSwerveModule', or something like that.

File: src/main/java/frc/robot/subsystems/MAXSwerveModule.java [hunk 4]
```diff
@@ -163,6 +163,7 @@ public void resetEncoders() {
     m_drivingEncoder.setPosition(0);
   }
 
+  //this function sets the idle mode of all the sparkmax motors to either brake or coast
```
Comment (noah4546) [inline src/main/java/frc/robot/subsystems/MAXSwerveModule.java]: Same issue as above, add space and capitalize the 'T' in 'This'

**Review by noah4546 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 3 above)
Comment (Stevon2386) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:252]: Our code is a fork of REV Robotics swerve code, and both REV and WPILib use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation) for their variables (Ex: m_ to mark it is a member variable, k to mark it as a constant. If we are getting rid of Hungarian notation we need to remove the m_ and k prefix from a large amount variables in our code.

Comment (noah4546) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:252]: Ok, I'll look into it, but still in java I believe we need to omit the under_score, if need fix, will be new PR.

**Review by noah4546 — APPROVED** [review]

**Review by dylanha06 — APPROVED** [review]

**Pull request merged**